src/main/java/md/jack/task9/Primes.java

```diff
@@ -0,0 +1,151 @@
+package md.jack.task9;
+
+import java.math.BigInteger;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.stream.Stream;
+
+/**
+ * A few utilities to generate a large random BigInteger,
+ * and find the next prime number above a given BigInteger.
+ * <p>
+ * From <a href="http://courses.coreservlets.com/Course-Materials/">the
+ * coreservlets.com tutorials on JSF 2, PrimeFaces, Ajax, JavaScript, jQuery, GWT, Android,
+ * Spring, Hibernate, JPA, RESTful Web Services, Hadoop, Spring MVC,
+ * servlets, JSP, Java 8 lambdas and streams (for those that know Java already),
+ * and Java 8 programming (for those new to Java)</a>.
+ */
+
+public class Primes
+{
+    private static final BigInteger ZERO = BigInteger.ZERO;
+    private static final BigInteger ONE = BigInteger.ONE;
+    private static final BigInteger TWO = new BigInteger("2");
+
+    // Likelihood of false prime is less than 1/2^ERR_VAL.
+    // Presumably BigInteger uses the Miller-Rabin test or
+    // equivalent, and thus is NOT fooled by Carmichael numbers.
+    // See Cormen et al.'s Introduction to Algorithms for details.
+    private static final int ERR_VAL = 100;
+    private static final String[] DIGITS =
+            "0,1,2,3,4,5,6,7,8,9".split(",");
+    private static final String[] NON_ZERO_DIGITS =
+            "0,1,2,3,4,5,6,7,8,9".split(",");
+
+    private Primes()
+    {
+    } // Uninstantiatable class
+
+    public static boolean isPrime(BigInteger possiblePrime)
+    {
+        return (possiblePrime.isProbablePrime(ERR_VAL));
+    }
+
+    public static boolean isPrime(long possiblePrime)
+    {
+        return (isPrime(new BigInteger(Long.toString(possiblePrime))));
+    }
+
+    /** Finds the next prime number above a threshold. */
+
+    private static BigInteger nextPrime(BigInteger start)
```

Comment:
method args has to be declared as final, try to rewrite this method after declare *start* object as final